test/parallel/test-buffer-swap.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+require('../common');
+const assert = require('assert');
+
+const buf = Buffer([0x1, 0x2, 0x3, 0x4]);
```

Comment:
Please also make a buffer large enough to require walking the array. e.g.
```js
const buf_array = [];
for (var i = 1; i < 33; i++)
  buf_array.push(i);
const buf2 = Buffer(buf_array);
```

Paranoia, but i've had bugs pop up for me in the past when haven't tested if the API could walk forward properly.